### PR TITLE
Nuttx next cmake update

### PR DESCRIPTION
**Pull request by davids5**
## Updated to upstream nuttx somewhere north of 7.14 

-   Updated the px4fmu-v2 defconfig (the rest need to be done when can)
-   New spi and i2c interface points
-   New irg enable/disable not critical section (name change only)

Long term todos 
- [ ] Update all configs. (make oldconfig in build and hand merge back to defconfig)
- [ ] Merge master into this branch - 
   - [ ] V4 HW
   - [ ] travis
- [ ]  regression test then replace nuttx_next_cmake (nuttx_next_cmake_last is left for AB testing) 
- [ ] Add procfs to regain ps etal commands

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@Inspirati @kd0aij Would it be possible to test-fly this?

**Comment by Inspirati** [comment]
I can put my focus towards perfecting the MPU9250+magnetometer driver on
pause for the time being (it's been rather doing my head in this week
anyhow), and build up my only AUAV-X1 board onto either a quadcopter or
into a bixler fixed-wing (both only half built right now). No doubt either
option would take at least a few days to produce any meaningful results.
I'm interested in any feedback on what may be the best division of effort
here?

Better yet, is the branch building for the pixracer target as yet? - i've
at least with that board i've got a spare of the beta units.

**Comment by kd0aij** [comment]
@Inspirati Need any help with the magnetometer driver?
@LorenzMeier I just noticed this PR; should it be flyable on both fmuv2 and v4?

**Comment by LorenzMeier** [comment]
This doesn't have v4 support yet. But you can test on v2 hardware. We still need to do the rebase on master with this one.

**Comment by kd0aij** [comment]
oops, I had the wrong origin set for this copy of the repo and inadvertently pushed here.
I wasn't getting mavlink data on USB, and traced it to incorrect arg handling.
Artificial horizon is working in QGC, but parameter upload is not.

File: src/modules/mavlink/mavlink_main.cpp [hunk 1]
```diff
@@ -1448,17 +1448,6 @@ Mavlink::task_main(int argc, char *argv[])
...
-	/* the NuttX optarg handler does not
-	 * ignore argv[0] like the POSIX handler
-	 * does, nor does it deal with non-flag
-	 * verbs well. Remove the application
-	 * name and the verb.
-	 */
-	argc -= 3;
-	argv += 3;
-#endif
-
```
Comment (davids5) [inline src/modules/mavlink/mavlink_main.cpp]: @kd0aij 

Is 3 correct? After @Inspirati 's nsh parser fix I would have expected it to be the same as what is on master https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_main.cpp#L1501-L1509 . Did you try 2?

**Comment by kd0aij** [comment]
@davids5 an offset of 2 looks correct now.
https://github.com/PX4/Firmware/pull/4023

**Comment by LorenzMeier** [comment]
@kd0aij @Inspirati Is this test flown and good to go into the nuttx_next branch?

**Comment by kd0aij** [comment]
px4fmu-v2_default build
short manual mode flight test on X2/QAV250: http://logs.uaventure.com/view/2WVEYK29nc5ngViCbxo5Tm

seems to fly OK, cpu load 25%

**Comment by LorenzMeier** [comment]
Awesome, thanks! Getting closer to being able to deploy the update.

**Pull request merged**